block-layer/codecs/encrypted.md

```diff
@@ -47,25 +48,29 @@ payload. Since the initializing vector is a known size for each AES variant the
 format is simply the iv followed by the byte payload.
 
 ```
-| iv | bytes |
+| varint(cipher-multicodec) | varint(ivLength) | iv | bytes |
```

Comment:
There is an argument about using a custom block format because of it's size compared to e.g. DAG-CBOR.

The current proposal is `| varint(cipher-multicodec) | varint(ivLength) | iv | bytes |` with the currently choosen multicodecs that's 3 bytes overhead compared to the actual data.

As a custom format needs custom code anyway, you could just hard-code the lengths of the `iv` based on the cipher and use an enum (and not a multicodec for the cipher type), so you end up with 1 byte overhead.

If you would use DAG-CBOR, when using an enum (and not multicodecs) for the ciphers and using list representation the overhead would be 7 bytes.

I'd go for either the 1 byte overhead or codec independent (e.g. DAG-CBOR), but I don't have a strong opinion.

Reply:
An interesting point about needing custom code anyway. Although it depends on where in the stack this custom code lives. For example, if a new cipher is added you'd now have to change the codec code which is a little awkward. Say we added support for this codec in go-ipfs but had not added support for a new cipher. With the extra bytes some python or js client could happily fetch the data from go-ipfs and decrypt with their custom cipher without any problems, by removing those bytes now the user needs to send a PR to go-ipfs to modify the codec to support their cipher.